### PR TITLE
PF-2509: update PAO overwrites existing attributes

### PR DESCRIPTION
**Pull request by mikenev**
We were working from a blank slate and then calculating removes and adds. This change will existing set and remove/replace inputs into it as necessary.

### CODE REVIEW

File: service/src/main/java/bio/terra/policy/service/pao/PaoService.java [hunk 1]
```diff
@@ -237,13 +237,14 @@ public PolicyUpdateResult updatePao(
         updateMode);
 
     Pao targetPao = paoDao.getPao(targetPaoId);
-    PolicyInputs newAttributes = new PolicyInputs();
```
Comment (yuhuyoyo) [inline service/src/main/java/bio/terra/policy/service/pao/PaoService.java:240]: can we call it attributesToUpdate instead

File: service/src/test/java/bio/terra/policy/service/pao/PaoUpdateTest.java [hunk 2]
```diff
@@ -320,6 +328,32 @@ void linkSourceHierarchyTest_policesCombineCorrectly() throws Exception {
...
+        PolicyInput.createFromMap(
+            TERRA_NAMESPACE,
+            REGION_CONSTRAINT,
+            Collections.singletonMap(REGION_KEY, DEFAULT_REGION_NAME));
+    PolicyInput groupPolicy =
+        PolicyInput.createFromMap(
+            TERRA_NAMESPACE, GROUP_CONSTRAINT, Collections.singletonMap(GROUP_KEY, GROUP_NAME));
+
+    UUID paoId = PaoTestUtil.makePao(paoService, regionPolicy);
+    logger.info("paoId: {}", paoId);
```
Comment (yuhuyoyo) [inline service/src/test/java/bio/terra/policy/service/pao/PaoUpdateTest.java:343]: nit: make pao {} with region policy.

**Review by yuhuyoyo — APPROVED** [review]
just some nits for readibility.

**Review by ddietterich — APPROVED** [review]
LGTM

**Pull request merged**